### PR TITLE
=str Make ConcurrentAsyncCallback extends AtomicReference directly.

### PR DESCRIPTION
**Pull request by He-Pin**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 1]
```diff
@@ -1188,21 +1187,22 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
    * [[onStop]] puts class in `Completed` state
    * "Real world" calls of [[invokeWithFeedback]] always return failed promises for `Completed` state
    */
-  private final class ConcurrentAsyncCallback[T](handler: T => Unit) extends AsyncCallback[T] {
+  private final class ConcurrentAsyncCallback[T](handler: T => Unit)
+      extends AtomicReference[State[T]](NoPendingEvents)
+      with AsyncCallback[T] {
     import ConcurrentAsyncCallbackState._
-    private[this] val currentState = new AtomicReference[State[T]](NoPendingEvents)
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala:1193]: reduce the size of ConcurrentAsyncCallback

**Review by johanandren — APPROVED** [review]
LGTM, nice optimization

**Comment by hepin1989** [comment]
@patriknw Ping~

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**